s3_upload/utils/upload.py

```
@@ -399,36 +399,45 @@ def multi_core_upload(
-        for future in as_completed(concurrent_jobs):
-            # access returned output as each is returned in any order
-            try:
-                uploaded_files, failed_upload = future.result()
+    for future in as_completed(concurrent_jobs):
+        # access returned output as each is returned in any order
+        try:
+            uploaded_files, failed_upload = future.result()
+
+            all_uploaded_files = {**all_uploaded_files, **uploaded_files}
+            all_failed_upload.extend(failed_upload)
+        except Exception as exc:
```

Ruff (0.7.0)

Do not catch blind exception: `Exception`

(BLE001)